src/core/mac/mac.cpp

```diff
@@ -2387,6 +2387,12 @@ void Mac::SetCslPeriod(uint16_t aPeriod)
 
     Get<DataPollSender>().RecalculatePollPeriod();
 
+    if (GetCslPeriod() == 0)
```

Comment:
```suggestion
    if ((GetCslPeriod() == 0) || IsCslEnabled())
```
Small suggestion for code simplification, allowing to remove the second `EnableCsl()` call below.

Reply:
Originally it was this way.
However this is the think which causes 1.2 csl certification test to fail. Since we are sending extra Child update request. Which will push through [this](https://github.com/openthread/openthread/blob/main/tests/scripts/thread-cert/v1_2_test_csl_transmission.py#L94) ping request.

Reply:
I mean this:
```
    if ((GetCslPeriod() == 0) || IsCslEnabled())
    {
        IgnoreError(Get<Radio>().EnableCsl(GetCslPeriod(), Get<Mle::Mle>().GetParent().GetRloc16(),
                                           &Get<Mle::Mle>().GetParent().GetExtAddress()));
    }

    if (IsCslEnabled())
    {
        Get<Mle::Mle>().ScheduleChildUpdateRequest();
    }
```

Reply:
fixed